evidence/reactor.go

```diff
@@ -118,16 +118,18 @@ func (evR *Reactor) broadcastEvidenceRoutine(peer p2p.Peer) {
 			case <-evR.Quit():
 				return
 			}
+		} else if !peer.IsRunning() || !evR.IsRunning() {
+			return
```

Comment:
Is this much the same as checking the `evR.Quit()` and `peer.Quit()` channels?

Reply:
yes, but we need it because `next != nil` sometimes.

Reply:
So how was the go routine not closing beforehand if we were checking for the `evr.Quit()` and `peer.Quit()` channels

Reply:
Good question. Does `peer.Send(EvidenceChannel, msgBytes)` block indefinitely if the peer is dropped?

Reply:
> So how was the go routine not closing beforehand if we were checking for the evr.Quit() and peer.Quit() channels

we weren't checking Quit because next wasn't NIL

> Does peer.Send(EvidenceChannel, msgBytes) block indefinitely if the peer is dropped?

no, it blocks for 10sec and then returns false.

Reply:
Ah right and it was never reaching the code at the end of the for loop where next gets set to nil because it was failing to send the evidence and thus the `continue` meant it would just remain in the loop